### PR TITLE
Add post submit routes for jotform, prevent duplicate submissions,

### PR DESCRIPTION
**Pull request by aoby**
add tests, and generally improve reliability.

This is the bulk of the work for JotForm improvements. Once this ships, I'll change all the forms in JotForm to redirect to our `submitRedirect` param and remove the existing environment-based conditional logic. 

There are also a few more work items coming up for the sync cronjob:
 1. separate into 2 jobs
    1. frequent sync that pulls answers for known placeholders
    2. less frequent (daily?) sanity check job that syncs everything
 2. add last successfully synced submission id to `Pd::SurveyQuestion` and update that once we've inspected a response (even if we reject due to wrong environment).
 3. Add check for missing environment (confirm that all answers are empty).

### CODE REVIEW

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 1]
```diff
@@ -16,16 +16,16 @@ class WorkshopDailySurveyController < ApplicationController
     # for the relevant session id.
     # The pre-workshop survey, which has no session id, will redirect to thanks.
     def new_general
+      # Accept days 0 through 4. Day 5 is the post workshop survey and should use the new_post route
+      day = params[:day].to_i
+      return render_404 if day < 0 || day > 4
```
Comment (mehalshah) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:21]: Should this ever happen? If this is pretty exceptional, I'd rather this be an error so we can see it in honey badger

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 2]
```diff
@@ -81,22 +106,49 @@ def new_facilitator
...
-        numFacilitators: facilitators.size
-      }
+        facilitatorName: facilitator.name,
+        numFacilitators: facilitators.size,
+        submitRedirect: url_for(action: 'submit_facilitator', params: {key: key_params})
+      )
+    end
+
+    # POST /pd/workshop_survey/facilitators/submit
+    def submit_facilitator
```
Comment (mehalshah) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:133]: This seems identical to the submit_general method. There seems like a DRY opportunity here keyed off whether facilitatorId is set

**Comment by aoby** [comment]
[JotForm improvements](https://trello.com/c/LBrXZ9kA/665-jotform-improvements)

File: dashboard/app/models/pd/workshop_daily_survey.rb [hunk 3]
```diff
@@ -81,5 +81,24 @@ def self.get_key_attributes(form_id, processed_answers)
         day: processed_answers['day'] || get_day_for_form_id(form_id)
       }
     end
+
+    def self.response_exists?(user_id:, pd_workshop_id:, day:, form_id:)
```
Comment (mehalshah) [inline dashboard/app/models/pd/workshop_daily_survey.rb:85]: Suggest having tests for these

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 1 above)
Comment (aoby) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:21]: Yes, absolutely. It's a URL: `/pd/workshop_survey/day/:day`. Anyone can just try another day and do something like https://studio.code.org/pd/workshop_survey/day/17. Why would we want to know about that? We should just say not found. That's a bad URL.

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb (hunk 2 above)
Comment (aoby) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:133]: Not so. The parameters are actually pretty different. I'll look to see if I can DRY it up a bit, but there are a lot of subtle differences, and I'd rather not make it obfuscated and complex in the guise of DRY.

Comment (aoby) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb:133]: And it's a different model. Different params, different model. The only common part is the first 3 lines. We could encapsulate that in a shared private method.

**Pull request merged**